components/compositing/windowing.rs

```diff
@@ -62,15 +62,20 @@ pub enum WindowEvent {
     /// Sent when the user resets zoom to default.
     ResetZoom,
     /// Sent when the user uses chrome navigation (i.e. backspace or shift-backspace).
-    Navigation(TraversalDirection),
+    Navigation(TopLevelBrowsingContextId, TraversalDirection),
     /// Sent when the user quits the application
     Quit,
     /// Sent when a key input state changes
     KeyEvent(Option<char>, Key, KeyState, KeyModifiers),
     /// Sent when Ctr+R/Apple+R is called to reload the current page.
-    Reload,
     /// Toggles the Web renderer profiler on and off
     ToggleWebRenderProfiler,
+    Reload(TopLevelBrowsingContextId),
+    /// Create a new top level browsing context
+    NewBrowser(ServoUrl, IpcSender<TopLevelBrowsingContextId>),
```

Comment:
Not yet. Ok with follow-up? I think this requires some more work in the constellation.